### PR TITLE
Fixed comparison of broken model enum classes

### PR DESCRIPTION
**Pull request by norberttech**
<h2>Change Log</h2>
<div id="change-log">
  <h4>Added</h4>
  <ul id="added">
    
  </ul> 
  <h4>Fixed</h4>  
  <ul id="fixed">
    <li>Fixed comparison of broken model enum classes</li>
  </ul>
  <h4>Changed</h4>
  <ul id="changed">
    
  </ul>  
  <h4>Removed</h4>
  <ul id="removed">
    
  </ul>
  <h4>Deprecated</h4>
  <ul id="deprecated">
    
  </ul>  
  <h4>Security</h4> 
  <ul id="security">
    
  </ul>     
</div>
<hr/>

<h2>Description</h2>

### CODE REVIEW

File: src/AmazonPHP/SellingPartner/ObjectSerializer.php [hunk 1]
```diff
@@ -345,12 +345,12 @@ public static function deserialize(Configuration $configuration, $data, string $
 
         if (\method_exists($class, 'getAllowableEnumValues')) {
             $brokenModelDefinitions = [
-                EventCode::class, // https://github.com/amazon-php/sp-api-sdk/issues/191
-                ItemImage::class, // https://github.com/amazon-php/sp-api-sdk/issues/156
+                \ltrim(EventCode::class, '/'), // https://github.com/amazon-php/sp-api-sdk/issues/191
```
Comment (Tetsuya-Takiguchi) [inline src/AmazonPHP/SellingPartner/ObjectSerializer.php:348]: @norberttech 
The string to be replaced is a backslash.
It is not a slash.

File: src/AmazonPHP/SellingPartner/ObjectSerializer.php [hunk 2]
```diff
@@ -345,12 +345,12 @@ public static function deserialize(Configuration $configuration, $data, string $
...
+                \ltrim(ItemImage::class, '/'), // https://github.com/amazon-php/sp-api-sdk/issues/156
```
Comment (Tetsuya-Takiguchi) [inline src/AmazonPHP/SellingPartner/ObjectSerializer.php:349]: The string to be replaced is a backslash.
It is not a slash.

File: src/AmazonPHP/SellingPartner/ObjectSerializer.php [hunk 3]
```diff
@@ -345,12 +345,12 @@ public static function deserialize(Configuration $configuration, $data, string $
...
             ];
 
             // Do not validate if class is one of amazon broken model definitions.
-            if (\in_array($class, $brokenModelDefinitions, true)) {
+            if (\in_array(\ltrim($class, '/'), $brokenModelDefinitions, true)) {
```
Comment (Tetsuya-Takiguchi) [inline src/AmazonPHP/SellingPartner/ObjectSerializer.php:353]: The string to be replaced is a backslash.
It is not a slash.

File: tests/AmazonPHP/SellingPartner/Tests/Unit/ObjectSerializerTest.php [hunk 4]
```diff
@@ -109,4 +110,19 @@ public function test_deserialize_label_format_enum_with_an_empty_string(): void
         $this->assertInstanceOf(ShippingServiceOptions::class, $object);
         $this->assertNull($object->getLabelFormat());
     }
+
+    public function test_serialization_of_enum_with_not_defined_value()
+    {
+        $object = ObjectSerializer::deserialize(
+            Configuration::forIAMUser('clientId', 'clientSecret', 'accessKey', 'secretKey'),
+            "EVENT_444",
+            \ltrim(EventCode::class, "/")
```
Comment (Tetsuya-Takiguchi) [inline tests/AmazonPHP/SellingPartner/Tests/Unit/ObjectSerializerTest.php:119]: The string to be replaced is a backslash.
It is not a slash.

File: tests/AmazonPHP/SellingPartner/Tests/Unit/ObjectSerializerTest.php [hunk 5]
```diff
@@ -109,4 +110,19 @@ public function test_deserialize_label_format_enum_with_an_empty_string(): void
...
+        );
+
+        $this->assertInstanceOf(EventCode::class, $object);
+        $this->assertEquals(
+            new EventCode("EVENT_444"),
+            $object
+        );
+    }
 }
```
Comment (Tetsuya-Takiguchi) [inline tests/AmazonPHP/SellingPartner/Tests/Unit/ObjectSerializerTest.php:128]: Please add a line break.

**Review by Tetsuya-Takiguchi — COMMENTED** [review]
@norberttech 
Typo confirmed.
We apologize for the inconvenience.

**Comment by Tetsuya-Takiguchi** [comment]
Hi. @norberttech 

https://github.com/norberttech/sp-api-sdk/blob/3e0421ca3b6934d5b3e5284e6613c6123ee36c87/src/AmazonPHP/SellingPartner/ObjectSerializer.php#L67

Sorry, I missed one point.
The same action needs to be taken at the location in the above link as in this case.

**Comment by norberttech** [comment]
Thanks @Tetsuya-Takiguchi, why don't you open a pull request? 
I don't have a reliable way to reproduce this behavior locally and test it against real data.

**Pull request merged**